File: backends/go-gin/internal/domain/invitation/auth_repository.go
// Package invitation は招待ドメインの型・インターフェースを提供します。
package invitation

// AuthRepository は招待認証トークンのキャッシュインターフェースです。
type AuthRepository interface {
	// Store はトークンとロールを指定秒数キャッシュします。
	Store(token string, role int, ttl int) error
	// Find はキャッシュからロールを取得します。存在しない場合は nil を返します。
	Find(token string) (*int, error)
	// Remove はトークンを削除します。
	Remove(token string) error
}

File: backends/ruby-rails/app/domain/invitation/auth_repository.rb
# frozen_string_literal: true
#
# 招待認証リポジトリインターフェースモジュール。
#
# @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

module Domain
  module Invitation
    # 招待認証トークンのキャッシュリポジトリインターフェースです。
    # @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
    module AuthRepository
      # @param token [String] 招待トークン
      # @param role [Integer] ロール（1=管理者, 2=メンバー）
      # @param ttl [Integer] キャッシュ有効期間（秒）
      def store(token, role, ttl) = raise NotImplementedError
      # @param token [String] 招待トークン
      # @return [Integer, nil] ロール値、または nil
      def find(token) = raise NotImplementedError
      # @param token [String] 招待トークン
      def remove(token) = raise NotImplementedError
    end
  end
end
